### PR TITLE
104-webchat-as-react-component

### PR DESCRIPTION
**Pull request by gabriel0003**
## Proposed changes

Web chat react component

## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Screenshots or links to design files

Does this change impact front end code?

- [ ] No
- [ ] Yes (please include screenshots or Figma files)

## Checklist

- [x] Code compiles correctly
- [ ] I have added tests that prove my fix is effective, or that my feature works
- [ ] Unit tests pass locally with my changes
- [ ] I have added/updated all necessary documentation
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: channels-api/config/env_config.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ type Config struct {
 		ServerAddress string `env:"CHANNELS_API_SERVER_ADDRESS,required"`
 		GRPCPort      int    `env:"CHANNELS_GRPC_PORT,required"`
 		OasisApiUrl   string `env:"OASIS_API_URL,required"`
+		CorsUrl       string `env:"CHANNELS_API_CORS_URL,required"`
```
Comment (tsearle) [inline channels-api/config/env_config.go:9]: not used (cors set to *)
delete un-used manadatory parameter

File: channels-api/config/env_config.go [hunk 2]
```diff
@@ -14,4 +15,8 @@ type Config struct {
 		SMTPFromUser      string `env:"SMTP_FROM_USER,required"`
 		ApiKey            string `env:"MAIL_API_KEY,required"`
 	}
+
+	WebChat struct {
+		ApiKey string `env:"WEBCHAT_API_KEY,required"`
```
Comment (tsearle) [inline channels-api/config/env_config.go:20]: document in README

File: channels-api/routes/webchat.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	RawMessage string `json:"rawMessage"`
+	Subject    string `json:"subject"`
+	ApiKey     string `json:"api-key"`
+}
+
+func addWebChatRoutes(conf *c.Config, rg *gin.RouterGroup) {
+
+	rg.GET("/webchat/", func(c *gin.Context) {
+		email := c.Query("email")
+		if err := c.ShouldBindUri(&email); err != nil {
```
Comment (tsearle) [inline channels-api/routes/webchat.go:42]: this line seems to be a bug...
1. you don't have a path parameter defined
2. you immediately are discarding the uri parameter you got

File: deployment/k8s/local-minikube/apps-config/channels-api.yaml [hunk 4]
```diff
@@ -40,3 +40,7 @@ spec:
               value: "demo@example.com"
             - name: MAIL_API_KEY
               value: "nobody-will-guess-this-api-key"
+            - name: CHANNELS_API_CORS_URL
```
Comment (tsearle) [inline deployment/k8s/local-minikube/apps-config/channels-api.yaml:43]: To Delete (not used)

File: packages/components/react/webchat/src/components/WebChatWindow/ChatEngine.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            console.log(stringify)
+        }).catch(error => {
+            console.log(error);
+        });
+    }
+
+    return (
+        <div style={{...styles.divStyle}}>
+            <MessageList
+                referance={messageListReferance}
```
Comment (edifirut) [inline packages/components/react/webchat/src/components/WebChatWindow/ChatEngine.tsx:82]: messageListReferance is not used. please remove it

File: packages/components/react/webchat/src/components/WebChatWindow/ChatEngine.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                className='message-list'
+                toBottomHeight={'100%'}
+                dataSource={datasource}
+                lockable={true}
+                downButton={false}
+                downButtonBadge={10}
+                sendMessagePreview={true}
+            />
+            <Input
+                referance={inputReferance}
```
Comment (edifirut) [inline packages/components/react/webchat/src/components/WebChatWindow/ChatEngine.tsx:92]: inputReferance is not used. please remove it

File: packages/components/react/webchat/src/components/WebChatWindow/ChatEngine.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        }).then(response => {
+            let stringify = JSON.stringify(response);
+            setDatasource((prevDatasource: any) => [...prevDatasource, {
+                position: 'right',
+                type: 'text',
+                text: msg,
+                date: new Date(),
+            }]);
+            clearRef();
+            forceUpdate();
```
Comment (edifirut) [inline packages/components/react/webchat/src/components/WebChatWindow/ChatEngine.tsx:72]: why we need this. there is no usage of "value" state param

Comment (gabriel0003) [reply packages/components/react/webchat/src/components/WebChatWindow/ChatEngine.tsx:72]: In order to clear the input after you press enter. Same for the other 2 comments

File: channels-api/config/env_config.go (hunk 1 above)
Comment (gabriel0003) [reply channels-api/config/env_config.go:9]: Cors is now set to this variable's value

File: channels-api/routes/main.go [hunk 8]
```diff
@@ -1,22 +1,38 @@
...
 		log.Fatalf("could not run server: %v", err)
 	}
 }
 
-func getRouter(conf *c.Config) *gin.Engine {
+func getRouter(conf *c.Config, fh *hub.WebChatMessageHub) *gin.Engine {
 	router := gin.New()
+	corsConfig := cors.DefaultConfig()
+
+	corsConfig.AllowOrigins = []string{conf.Service.CorsUrl}
```
Comment (tsearle) [inline channels-api/routes/main.go:23]: change to 
corsConfig.AllowOrigins = strings.Split(conf.Service.CorsUrl, " ")
to allow multiple domains and make Jonty happy
link
https://github.com/openline-ai/openline-oasis/issues/101
to this branch

**Review by tsearle — APPROVED** [review]

**Pull request merged**